lib/Doctrine/Hook.php

```diff
@@ -157,11 +157,11 @@ public function hookWhere($params)
                     if (isset($this->typeParsers[$def[0]])) {
                         $name   = $this->typeParsers[$def[0]];
                         $parser = new $name;
-                    }
 
-                    $parser->parse($alias, $column, $value);
```

Comment:
Kind of hard to tell from this diff, but these two lines were moved up into the `if` condition where `$parser` gets defined.  It didn't make much sense to call methods on `$parser` if parser is not defined.